Creé este archivo para cubrir algunos errores que no habia tenido en cuenta.

importamos las librerias

In [1]:
import pandas as pd
import numpy as np

importamos el archivo al cual queremos modificar

In [2]:
tec_provincias = pd.read_csv('nuevos_datos/tecnologia_provincias.csv')

creamos una copia

In [4]:
tec_copy = tec_provincias.copy()

eliminamos duplicados basandonos en año

In [5]:
tec_copy.drop_duplicates(subset='Año')

Año Trimestre     Provincia   ADSL  Cablemodem  Fibra óptica  \
0      2022         4  Buenos Aires  326.0         3.0           1.0   
96     2021         4  Buenos Aires  426.0         3.0           1.0   
192    2020         4  Buenos Aires  636.0         2.0         805.0   
288    2019         4  Buenos Aires  720.0         2.0         669.0   
312  2019 *       3 *  Buenos Aires    1.0         2.0         425.0   
384    2018         4  Buenos Aires    1.0         2.0         296.0   
480    2017         4  Buenos Aires    1.0         2.0         131.0   
576    2016         4  Buenos Aires    2.0         1.0         119.0   
672    2015         4  Buenos Aires    2.0         1.0         103.0   
768    2014         4  Buenos Aires    2.0         1.0         121.0   

     Wireless  Otros  indice  
0       130.0   72.0       0  
96      127.0   73.0      96  
192      93.0   58.0     192  
288      87.0   18.0     288  
312      86.0   27.0     312  
384      65.0   20.0     384  
480      60.0   31.0     480  
576      21.0   24.0     576  
672      20.0   26.0     672  
768      18.0   32.0     768

como vemos en el indice 312 hay un error de carga, el cual pone un * al final, lo que hare sera cambiar eso por el año correspondiente

In [7]:
tec_provincias[tec_provincias['Año'] == '2019 *'] = 2019

ahora haremos lo mismo con Trimestre

In [8]:
tec_copy.drop_duplicates(subset='Trimestre')

Año Trimestre     Provincia   ADSL  Cablemodem  Fibra óptica  \
0      2022         4  Buenos Aires  326.0         3.0           1.0   
24     2022         3  Buenos Aires  340.0         3.0           1.0   
48     2022         2  Buenos Aires  351.0         3.0           1.0   
72     2022         1  Buenos Aires  392.0         3.0           1.0   
312  2019 *       3 *  Buenos Aires    1.0         2.0         425.0   
336  2019 *       2 *  Buenos Aires    1.0         2.0         383.0   
360  2019 *       1 *  Buenos Aires    1.0         2.0         336.0   

     Wireless  Otros  indice  
0       130.0   72.0       0  
24      127.0   70.0      24  
48      127.0   71.0      48  
72      126.0   67.0      72  
312      86.0   27.0     312  
336      75.0   33.0     336  
360      65.0   20.0     360

Aqui vemos que en el indice 312, 336 y 360 nos muestra los trimestres mal cargados, lo que hare sera cambiarlos como hice con año.

antes de eso convertiremos la columna en un str

In [17]:
tec_provincias['Trimestre'] = tec_provincias['Trimestre'].astype(str)

filtramos en busca de los errores

In [31]:
tec_provincias[(tec_provincias['Trimestre'] != '1') & (tec_provincias['Trimestre'] != '2') & (tec_provincias['Trimestre'] != '3') & (tec_provincias['Trimestre'] != '4')]

Año Trimestre Provincia    ADSL  Cablemodem  Fibra óptica  Wireless  \
312  2019      2019      2019  2019.0      2019.0        2019.0    2019.0   
313  2019      2019      2019  2019.0      2019.0        2019.0    2019.0   
314  2019      2019      2019  2019.0      2019.0        2019.0    2019.0   
315  2019      2019      2019  2019.0      2019.0        2019.0    2019.0   
316  2019      2019      2019  2019.0      2019.0        2019.0    2019.0   
..    ...       ...       ...     ...         ...           ...       ...   
379  2019      2019      2019  2019.0      2019.0        2019.0    2019.0   
380  2019      2019      2019  2019.0      2019.0        2019.0    2019.0   
381  2019      2019      2019  2019.0      2019.0        2019.0    2019.0   
382  2019      2019      2019  2019.0      2019.0        2019.0    2019.0   
383  2019      2019      2019  2019.0      2019.0        2019.0    2019.0   

      Otros  indice  
312  2019.0    2019  
313  2019.0    2019  
314  2019.0    2019  
315  2019.0    2019  
316  2019.0    2019  
..      ...     ...  
379  2019.0    2019  
380  2019.0    2019  
381  2019.0    2019  
382  2019.0    2019  
383  2019.0    2019  

[72 rows x 9 columns]

aqui vemos una serie de errores de carga. lo que hare sera eliminar los errores atravez del indice

obtenemos el indice

In [35]:
indices = tec_provincias[tec_provincias['Trimestre'] == '2019'].index

creamos un for para eliminarlos

In [36]:
for i in indices:
    tec_provincias.drop(index=i,inplace=True)

cambiamos el tipo de dato

In [38]:
tec_provincias['Trimestre'] = tec_provincias['Trimestre'].astype(int)

y cargamos nuevamente los archivos

In [39]:
tec_provincias.to_csv('nuevos_datos/tecnologia_provincias.csv',index=False)